eng/common/pipelines/templates/steps/verify-links.yml

```diff
@@ -2,13 +2,28 @@ parameters:
   Directory: 'not-specified'
   IgnoreLinksFile: "$(Build.SourcesDirectory)/eng/ignore-links.txt"
   
-
 steps:
-  - task: PowerShell@2
-    displayName: Link verification check
-    inputs:
-      pwsh: true
-      workingDirectory: $(Build.SourcesDirectory)/${{ parameters.Directory }}
-      filePath: eng/common/scripts/Verify-Links.ps1 
-      arguments: >
-        -urls $(dir -r -i *.md) -rootUrl "file://$(Build.SourcesDirectory)/${{ parameters.Directory }}" -recursive:$false -ignoreLinksFile ${{ parameters.IgnoreLinksFile }}
+- pwsh: |
+    try {
+      echo "##vso[task.setvariable variable=replacedBranch]$(system.pullRequest.sourceCommitId)"
+    } catch {
+      echo "##vso[task.setvariable variable=replacedBranch]''"
+      echo "It is not a pr triggered build."
+    }
+
+  displayName: Triggered by PR check
+  workingDirectory: ${{ parameters.WorkingDirectory }}
+  ignoreLASTEXITCODE: true
+- task: PowerShell@2
+  displayName: Link verification check
+  inputs:
+    pwsh: true
+    workingDirectory: $(Build.SourcesDirectory)/${{ parameters.Directory }}
+    filePath: eng/common/scripts/Verify-Links.ps1 
+    arguments: >
+      -urls $(dir -r -i *.md) 
+      -rootUrl "file://$(Build.SourcesDirectory)/${{ parameters.Directory }}" 
+      -recursive $false 
+      -ignoreLinksFile ${{ parameters.IgnoreLinksFile }}
+      -branchReplaceRegex "(https://github.com/.*/blob/)master(/.*)"
+      -branchReplacementName $(replacedBranch)
```

Comment:
Here is another potential option try using `$env:SYSTEM_PULLREQUEST_SOURCECOMMITID` https://docs.microsoft.com/en-us/azure/devops/pipelines/process/variables?view=azure-devops&tabs=yaml%2Cbatch#access-variables-through-the-environment. I think those will be empty if they don't have a value.

Reply:
Will try